### PR TITLE
colord: add enableSystemd option

### PR DESCRIPTION
**Pull request by alyssais**
Since https://github.com/NixOS/nixpkgs/pull/345979, building colord is necessary to build xdg-desktop-portal-gtk, but it wasn't possible for systems that can't use systemd.

I've called the option "enableSystemd" for consistency with the existing "enableDaemon" option.

As usual in Nixpkgs, the user is responsible for also overriding udev
to a non-systemd implementation if necessary.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/colord/default.nix [hunk 1]
```diff
@@ -60,6 +62,8 @@ stdenv.mkDerivation rec {
     "-Dvapi=true"
     "-Ddaemon=${lib.boolToString enableDaemon}"
     "-Ddaemon_user=colord"
+    (lib.mesonBool "systemd" enableSystemd)
+    (lib.mesonBool "udev_rules" (lib.elem "udev" udev.meta.pkgConfigModules))
```
Comment (K900) [inline pkgs/tools/misc/colord/default.nix:66]: Ooh I don't like this

Comment (alyssais) [reply pkgs/tools/misc/colord/default.nix:66]: Not all udev implementations (libudev-zero) implement udev rules, just the library interface.  To indicate that, they don't provide udev.pc, which is a file that only tells you the udev rules directory.  In that case, udev_rules must be disabled, or the build will fail due to it not being able to tell where to install its rules due to lack of udev.pc.  You might think that we could override that using pkg-config environment variables, but unfortunately it only looks at those _after_ it's found the .pc files.

Comment (K900) [reply pkgs/tools/misc/colord/default.nix:66]: Wow this is cursed.

**Review by jtojnar — APPROVED** [review]
Looks alright to me, thanks.

**Comment by alyssais** [comment]
@OfBorg build eudev.tests.pkg-config libudev-zero.tests.pkg-config systemd.tests.pkg-config

**Pull request merged**